GameServer/Finance/Currency.cs

```diff
@@ -1,102 +1,66 @@
 using System;
-using System.Collections.Generic;
-using DOL.Database;
 
 namespace DOL.GS.Finance
 {
     public abstract class Currency
     {
-        protected Currency() { }
+        public virtual string Name { get; }
+        public bool IsItemCurrency => this is ItemCurrency;
 
-        public static Currency Create(byte currencyId, string itemTemplateId = null)
-        {
-            switch (currencyId)
-            {
-                case 1: return Copper;
-                case 2: return ItemCurrency.CreateFromItemTemplateId(itemTemplateId);
-                case 3: return BountyPoints;
-                case 4: return Mithril;
-                default: throw new System.NotImplementedException($"Currency with id {currencyId} is not implemented.");
-            }
-        }
+        protected Currency() { }
 
-        public static Currency Copper { get; } = new Copper();
-        public static Currency BountyPoints { get; } = new BountyPoints();
-        public static Currency Mithril { get; } = new Mithril();
-        public static Currency Item(ItemTemplate itemTemplate) => ItemCurrency.Create(itemTemplate);
-        public static Currency Item(string itemTemplateId) => ItemCurrency.CreateFromItemTemplateId(itemTemplateId);
+        public static Currency Copper { get; } = new CopperCurrency();
+        public static Currency BountyPoints { get; } = new BountyPointsCurrency();
+        public static Currency Mithril { get; } = new MithrilCurrency();
+        public static Currency Item(string currencyId) => new ItemCurrency(currencyId);
 
         public Money Mint(long value) => Money.Mint(value, this);
 
-        public abstract string Name { get; }
+        public virtual string ToText() => Name;
```

Comment:
Localization is intentionally omitted here as localization needs a rework anyway and is currently done only partly at best.

Reply:
I agree for the localization refactoring/rework, I was suggesting that because it's not really linked to the locales but more to have everything we should need for a good localization.
Maybe you are suggesting to have only one language for the whole server, in this case my suggestion are completely useless :)

Reply:
Localization was a genuine afterthought here. It's clear to me that localization should not find the way into any `internal`/`public` method/ctor signature, not even indirectly via GameClient, however I don't have a solution for solving the localization issue at hand yet and I won't tackle it immediately. 

My stance here is, that localization is a sensible feature, but it's still rather low on my metaphorical priority list.